data-extraction/pyDataExtraction/commonTypes/Graph.py

```diff
@@ -0,0 +1,76 @@
+from typing import Union, Dict, Optional
+from pyDataExtraction.commonTypes.base import DataType
+
+# from docs
+"""
+interface NodeGraphData {
+	id: string;
+	label?: string;
+	color?: string;
+	shape?: "ellipse" | "box";
+}
+"""
+"""
+interface EdgeGraphData {
+	from: string;
+	to: string;
+	label?: string;
+	id?: string;
+	color?: string;
+	dashes?: boolean;
+}
+"""
+# NOTE: may not be able to encapsulate edge in separate class due to from being a syntax token
+"""
+class Edge:
+    
+    def __init__(self,from: str, to: str,):
+        self.fromnode
+        
+    def __repr__(self):
+        pass
+"""
+# NOTE: ran into issue Node object is not json serializable when ecapsulating in own class
+"""
+class Node(DataType):
+    def __init__(self, id: Union[int, str], label: Optional[str] = None):
+        super().__init__()
+        self.id = id
+        if label is None:
```

Comment:
Suggest simplifying to:
`self.lable = lable or id`

Reply:
that is actually a hella useful feature I didn't know about. Thanks for that.

Reply:
also, given the implementation, and what we are passing the information to, this likely wouldn't work, as the attribute would still be listed in the json output(at least with the method currently used to produce json.

I'm not sure if this would be the case, but I'm trying to avoid all of the nodes being labeled `None` in the visualizer. So, I set it up to where attributes were only present in the case where a value was assigned.

Reply:
Can you give an example? I'm not able to understand.
As from what I think if we go with
`self.label = label or id`
If anyone doesn't pass `label` the default is `None` and the `self.label` will be populated by `id`.
Correct?